### PR TITLE
Spark proposals

### PR DESCRIPTION
**Pull request by Blady-Com**
Hello,
Some change proposals that might ease proof process ;-)
Pascal.

### CODE REVIEW

**Review by jcdubois — APPROVED** [review]
Thanks, these changes are making the code a lot easier to read.
All these explicit cast were bothering me.

**Review by jcdubois — APPROVED** [review]
Good, this patch avoid to have to implement/link the __ashldi3 function.

**Review by jcdubois — COMMENTED** [review]
For now the introduction of this patch (the second one) is triggering an exception and a crash in the OS.

I need to investigate this.

File: kernel/core/include/os.ads [hunk 1]
```diff
@@ -63,8 +63,7 @@ is
    subtype os_status_t is types.int32_t
                            range OS_ERROR_MAX .. OS_SUCCESS;
 
-   subtype os_mbx_index_t is types.uint8_t
-                           range 0 .. OS_MAX_MBX_ID;
+   type os_mbx_index_t is mod OS_MAX_MBX_ID;
```
Comment (jcdubois) [inline kernel/core/include/os.ads:66]: Does moving to a modular type keeps the 8 bits (char) size of the type?

Comment (Blady-Com) [reply kernel/core/include/os.ads:66]: Not necessary, it's up to the compiler.
To force it, add:
for os_mbx_index_t'size use 8;
or with aspect flavor:
type os_mbx_index_t is mod OS_MAX_MBX_ID
   with Size => 8;

Comment (jcdubois) [reply kernel/core/include/os.ads:66]: I resolved the problem by splitting the os_task_rw element in 2 parts.
- one part dedicated to the C part and unknown to the Ada part
- one part dedicated to the Ada part and unknown from the C part.

This way we can let Ada optimize the data size as it see fit without requiring data compatibility with the C part.

**Pull request merged**